patches/extensions-browser-api-socket-tcp_socket.cc.patch

```diff
@@ -0,0 +1,17 @@
+diff --git a/extensions/browser/api/socket/tcp_socket.cc b/extensions/browser/api/socket/tcp_socket.cc
+index a1ee755366f7a35917735fe411123d44450111ed..017ef2940e6530be7939a80e7a34e0b7fa28d1df 100644
+--- a/extensions/browser/api/socket/tcp_socket.cc
++++ b/extensions/browser/api/socket/tcp_socket.cc
+@@ -118,8 +118,11 @@ void TCPSocket::Disconnect(bool socket_destroying) {
+   if (socket_.get())
+     socket_->Disconnect();
+   server_socket_.reset(NULL);
+-  connect_callback_.Reset();
+   // TODO(devlin): Should we do this for all callbacks?
++  if (!connect_callback_.is_null()) {
++    base::ResetAndReturn(&connect_callback_)
++        .Run(net::ERR_CONNECTION_CLOSED);
```

Comment:
Solve DCHECK erorr at https://cs.chromium.org/chromium/src/extensions/browser/extension_function.cc?sq=package:chromium&g=0&l=504.
Webtorrent hit the DCHECK where SocketsTcpConnectFunction is destructed because it is possible for the application to make a request to destroy the socket while previous connect request is still in progress.
This DCHECK is introduced by https://codereview.chromium.org/2017113002, in the same changeset, read_callback is reset and return immediately here.

Reply:
This is outdated, it only works for C69, I'll remove it